devito/types/tensor.py

```diff
@@ -1,19 +1,20 @@
 from collections import OrderedDict
 
-import sympy
 import numpy as np
-from sympy.core.sympify import converter as sympify_converter
+from sympy import Dummy
 from sympy.core.decorators import call_highest_priority
+from sympy.core.sympify import converter as sympify_converter
 
 from devito.finite_differences import Differentiable
-from devito.types.basic import AbstractTensor, Basic
+from devito.tools import flatten
+from devito.types.basic import AbstractTensor
 from devito.types.dense import Function, TimeFunction
 from devito.types.utils import NODE
 
 __all__ = ['TensorFunction', 'TensorTimeFunction', 'VectorFunction', 'VectorTimeFunction']
 
 
-class TensorFunction(AbstractTensor, Differentiable):
+class TensorFunction(AbstractTensor):
```

Comment:
No more Differentiable,  but below it still uses

```
_op_priority = Differentiable._op_priority + 1.
```

that's quite odd

Reply:
This is just for priority reasons, so that Tensor Mul is higher priority than Differentiable. I could just put the value to avoid the "dependency" but feels clearer with it.

Reply:
Btw, remind me why not Differentiable anymore?

Reply:
Thought wrote it somewhere. The components are differentiable in a functional analysis sense, we don't support matricial differential operation for say.